### PR TITLE
add more security check to dangerous operations

### LINKED ISSUE #98: Restrict dog permissions
Tested on debian, sheepdog v 0.82.

Dog can be run by any user just typing the full path /usr/sbin/dog.
This isn't wrong, but a normal user is able to create and delete vdi.
I guess it's able even to format the cluster.

Dog should restrict dangerous operation only to root.

vdi check
vdi create
vdi snapshot
vdi clone
vdi delete
vdi rollback
vdi setattr
vdi getattr
vdi resize
vdi read
vdi write
vdi backup
vdi restore
vdi cache
node kill
node md
node log
cluster format
cluster shutdown
cluster snapshot
cluster recover
cluster reweight
cluster check

Others subcommands may be run as normal user.
This may be useful for monitoring the cluster without requiring super user privilege.

vdi list
vdi tree
vdi graph
vdi object
vdi track
node list
node info
node recovery
node stat
cluster info

### PR DESCRIPTION
**Pull request by vtolstov**
fixes #98

Signed-off-by: Vasiliy Tolstov <v.tolstov@selfip.ru>

### CODE REVIEW

**Comment by mitake** [comment]
seems nice, but it caused build error:
```
common.c: In function ‘do_generic_subcommand’:
common.c:293:4: warning: implicit declaration of function ‘is_root’ [-Wimplicit-function-declaration]
    if (!is_root()) {
    ^
common.c: At top level:
common.c:544:6: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
 bool is_root()
      ^
common.c:544:6: error: conflicting types for ‘is_root’
common.c:293:9: note: previous implicit declaration of ‘is_root’ was here
    if (!is_root()) {
         ^
```

cc @sirio81

**Comment by vtolstov** [comment]
compile errors fixed

**Comment by vtolstov** [comment]
any problems with patch?

File: dog/common.c [hunk 1]
```diff
@@ -533,3 +540,11 @@ uint8_t parse_copy(const char *str, uint8_t *copy_policy)
 	copy = copy + parity;
 	return copy;
 }
+
+bool is_root()
```
Comment (mitake) [inline dog/common.c]: bool is_root(void)

**Comment by mitake** [comment]
Thanks for your updates. Some comments:
- please exclude merge master commit
- please squash the commits into single one
- please change a type of the function to bool is_root(void) (not is_root())
- you also need to add a check to main() of dog command. here would be a suitable place: https://github.com/sheepdog/sheepdog/blob/master/dog/dog.c#L557

**Comment by vtolstov** [comment]
@mitake 
```
you also need to add a check to main() of dog command. here would be a suitable place: https://github.com/sheepdog/sheepdog/blob/master/dog/dog.c#L557
```

sorry i'm not understand, what check i need to add? if i check for root in dog, and return, i can't run unprivileged subcommands...
Or i miss something?

**Comment by vtolstov** [comment]
i'm squash commits and fix

**Comment by mitake** [comment]
pushed with trivial commit log fix, thanks!

**Pull request closed without merging**